### ISSUE TITLE
Apply with custom DataLoader

### ISSUE THREAD

**Issue by leviethung2103** [mjkvaak/ImageDataAugmentor#5]
Hello, 
I am very impressed with your work. 
Currently, I am working with custom Data Augmentation. As you mentioned in your repo, the ImageDataAugmentor can work properly with a specific methods such as `.flow(x,y)` or `from_directory`. 

How to integrate with another custom data loader that I've already defined. It just class DataLoader that yields a batch of samples. 

Thank you for reading question.

**Comment by leviethung2103** [comment]
Here is the sample snippet code for Data Loader Class
`import numpy as np
import keras

class DataGenerator(keras.utils.Sequence):
    'Generates data for Keras'
    def __init__(self, list_IDs, labels, batch_size=32, dim=(32,32,32), n_channels=1,
                 n_classes=10, shuffle=True):
        'Initialization'
        self.dim = dim
        self.batch_size = batch_size
        self.labels = labels
        self.list_IDs = list_IDs
        self.n_channels = n_channels
        self.n_classes = n_classes
        self.shuffle = shuffle
        self.on_epoch_end()

    def __len__(self):
        'Denotes the number of batches per epoch'
        return int(np.floor(len(self.list_IDs) / self.batch_size))

    def __getitem__(self, index):
        'Generate one batch of data'
        # Generate indexes of the batch
        indexes = self.indexes[index*self.batch_size:(index+1)*self.batch_size]

        # Find list of IDs
        list_IDs_temp = [self.list_IDs[k] for k in indexes]

        # Generate data
        X, y = self.__data_generation(list_IDs_temp)

        return X, y

    def on_epoch_end(self):
        'Updates indexes after each epoch'
        self.indexes = np.arange(len(self.list_IDs))
        if self.shuffle == True:
            np.random.shuffle(self.indexes)

    def __data_generation(self, list_IDs_temp):
        'Generates data containing batch_size samples' # X : (n_samples, *dim, n_channels)
        # Initialization
        X = np.empty((self.batch_size, *self.dim, self.n_channels))
        y = np.empty((self.batch_size), dtype=int)

        # Generate data
        for i, ID in enumerate(list_IDs_temp):
            # Store sample
            X[i,] = np.load('data/' + ID + '.npy')

            # Store class
            y[i] = self.labels[ID]

        return X, keras.utils.to_categorical(y, num_classes=self.n_classes)`

**Comment by mjkvaak** [comment]
Maybe I missed what you're asking (apologies for that) but as I understood, you would like to add the `.flow` and `flow_from_directory` functionalities to your own script. In `ImageDataAugmentor` this has been solved by two additional classes, [`NumpyArrayIterator`](https://github.com/mjkvaak/ImageDataAugmentor/blob/master/ImageDataAugmentor/numpy_array_iterator.py) and [`DirectoryIterator`](https://github.com/mjkvaak/ImageDataAugmentor/blob/master/ImageDataAugmentor/directory_iterator.py) respectively. 

I'm afraid that I don't have any simple solution to add the functionalities to your `DataGenerator`. You may want to mimic the script in the classes that I attached above, or if you want to make it easier for yourself, write two separate `DataGenerator` classes that have different parameter inputs, e.g. `DataGenerator1` with `x, y` and `DataGenerator2` with `data_dir`.

Hope this helps. I will keep the issue open in case you want to discuss this further.

**Comment by mjkvaak** [comment]
Closing due to inactivity.
